en/development/testing.rst

```diff
@@ -1011,6 +1011,62 @@ assertions in `TestSuite
 found in `PHPUnit
 <https://phpunit.de/manual/current/en/appendixes.assertions.html>`__.
 
+Comparing test results to a file
+--------------------------------
+
+For some types of test, it may be easier to compare the result of a test to the
+contents of a file - for example, when testing the rendered output of a view.
+The ``StringCompareTrait`` adds a simple assert method for this purpose.
+
+Usage involves using the trait, setting the comparison base path and calling
+``assertSameAsFile``::
+
+    use Cake\TestSuite\StringCompareTrait;
+    use Cake\TestSuite\TestCase;
+
+    class SomeTest extends TestCase
+    {
+        use StringCompareTrait;
+
+        public function setUp()
+        {
+            $this->_compareBasePath = APP . 'tests/comparisons/';
+            parent::setUp();
+        }
+
+        public function testExample()
+        {
+            $result = ...;
+            $this->assertSameAsFile('example.php', $result);
```

Comment:
I think the "what about Windows" was mainly about the different PHP_EOL chars.
And that those should maybe have to be unified before comparison (not sure of the assertSameAsFile() wrapper already does that).